wgengine/router/router_windows.go

```diff
@@ -111,6 +111,13 @@ func (r *winRouter) Set(cfg *Config) error {
 		return fmt.Errorf("dns set: %w", err)
 	}
 
+	out, err := exec.Command("ipconfig", "/flushdns").CombinedOutput()
```

Comment:
I was thinking that this might need to go in wgengine/router/dns/manager_windows.go, next to the other ipconfig command `ipconfig /registerdns`, but no, it needs to be here because it has to run whenever the network config changes even if the DNS config doesn't change. So, this looks good.

Can you include a comment that links to the github issue, and say that we flush DNS in case we're dealing with a split horizon DNS server?

Reply:
Added a comment with the issue number, but pls check it - I haven't understood your "split horizon DNS server" part.

Reply:
Looks good.

The original bug we had was with enterprise customers using "split horizon" DNS, where DNS names resolve to public IPs when you're on the internet, and LAN IPs when you're on the office network. The customer wanted Tailscale to correctly resolve to internal IPs when they roam from public wifi to office wifi, so we have to flush DNS on network change.

Split horizon DNS is a bit of a problematic configuration in general because so much stuff assumes that DNS records don't change too much based on your network location... But it's a popular corporate configuration :(